clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -863,9 +863,9 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             AbstractResponse body = AbstractResponse.
                     parseResponse(req.header.apiKey(), responseStruct, req.header.apiVersion());
             maybeThrottle(body, req.header.apiVersion(), req.destination, now);
-            if (req.isInternalRequest && body instanceof MetadataResponse)
-                metadataUpdater.handleSuccessfulResponse(req.header, now, (MetadataResponse) body);
-            else if (req.isInternalRequest && body instanceof ApiVersionsResponse)
+            if (req.isInternalRequest && body instanceof MetadataResponse) {
+                metadataUpdater.handleSuccessfulResponse(req.header, now, (MetadataResponse) body, source);
```

Comment:
This source will be the integer form a broker's id. I think it's better to use the req.destination so that it's consistent with other logs, e.g. the log above
```
log.trace("Completed receive from node {} for {} with correlation id {}, received {}", req.destination,
                    req.header.apiKey(), req.header.correlationId(), responseStruct);
```